### PR TITLE
Refactoring tracing interfaces to `camel-api` and `camel-support`

### PR DESCRIPTION
**Pull request by ammachado**
This allows moving the span decorators closer to the components.

# Description

Move tracing span interfaces to `camel-api` and abstract implementations to `camel-support`. Refactoring the current span property as a `ExchangePropertyKey`/internal property.

# Target

- [x] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [x] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [x] I checked that each commit in the pull request has a meaningful subject line and body.
- [x] I have run `mvn clean install -DskipTests` locally from root folder and I have committed all auto-generated changes.

### CODE REVIEW

File: core/camel-api/src/main/java/org/apache/camel/Exchange.java [hunk 1]
```diff
@@ -67,6 +67,7 @@
 @ConstantProvider("org.apache.camel.ExchangeConstantProvider")
 public interface Exchange extends VariableAware {
 
+    String ACTIVE_SPAN = "OpenTracing.activeSpan";
```
Comment (orpiske) [inline core/camel-api/src/main/java/org/apache/camel/Exchange.java:70]: qq: this is not following the pattern we use. Is there a reason for that?

**Review by orpiske — CHANGES_REQUESTED** [review]
I'm not entirely sure I understand the motivations for the change, so marking as requested changes to wait for more feedback.

File: core/camel-api/src/main/java/org/apache/camel/Exchange.java (hunk 1 above)
Comment (ammachado) [reply core/camel-api/src/main/java/org/apache/camel/Exchange.java:70]: This was copied from `camel-trace`, but I agree that it should be renamed.

**Comment by ammachado** [comment]
> I'm not entirely sure I understand the motivations for the change, so marking as requested changes to wait for more feedback.

The idea of this pull request is to move functionality from `camel-trace` to the core libraries, and move all span decorators to their respective components.

File: core/camel-api/src/main/java/org/apache/camel/Exchange.java (hunk 1 above)
Comment (davsclaus) [reply core/camel-api/src/main/java/org/apache/camel/Exchange.java]: This key is also used in `core/camel-support/src/main/java/org/apache/camel/saga/InMemorySagaCoordinator.java`

**Comment by davsclaus** [comment]
This requires a JIRA for tracking something like this happening so users can see this in changelogs

**Comment by davsclaus** [comment]
The decorates can be moved to each component - however so far it has not been needed and having them in one place allows to easier identify which decorators we have. Also have you had a need for a decorator to rely directly on its component APIs vs as it is now.

**Review by oscerd — CHANGES_REQUESTED** [review]
I'm not sure what would be the advantages of having decorators directly in the components.. If we place all of them in a single point, we could easily know what we support and what we don't, even for catalog purpose. If we move them in the single components it would be much more complicated.

**Pull request closed without merging**